apps/web/pages/settings/my-account/profile.tsx

```diff
@@ -203,133 +179,100 @@ const ProfileView = () => {
     [ErrorCode.InternalServerError]: `${t("something_went_wrong")} ${t("please_try_again_and_contact_us")}`,
     [ErrorCode.ThirdPartyIdentityProviderEnabled]: t("account_created_with_identity_provider"),
   };
-  const onSuccessfulUsernameUpdate = async () => {
-    showToast(t("settings_updated_successfully"), "success");
-    await utils.viewer.me.invalidate();
-  };
-
-  const onErrorInUsernameUpdate = () => {
-    showToast(t("error_updating_settings"), "error");
-  };
 
-  if (isLoading || !user)
+  if (isLoading || !user || isLoadingAvatar || !avatar)
     return <SkeletonLoader title={t("profile")} description={t("profile_description")} />;
 
-  const isDisabled = isSubmitting || !isDirty;
+  const defaultValues = {
+    username: user.username || "",
+    avatar: avatar.avatar || "",
+    name: user.name || "",
+    email: user.email || "",
+    bio: user.bio || "",
+  };
 
   return (
     <>
       <Meta title={t("profile")} description={t("profile_description", { appName: APP_NAME })} />
-      <Form
-        form={formMethods}
-        handleSubmit={(values) => {
-          if (values.email !== user?.email && isCALIdentityProviver) {
+      <ProfileForm
+        key={JSON.stringify(defaultValues)}
```

Comment:
We trigger a re-render after mutation so form gets reinitialized and isDirty logic reapplies